src/shared/watchdog.c

```diff
@@ -115,8 +118,18 @@ int watchdog_set_timeout(usec_t *usec) {
 }
 
 int watchdog_ping(void) {
+        usec_t ntime;
         int r;
 
+        ntime = now(clock_boottime_or_monotonic());
+
+        if (_unlikely_(watchdog_last_ping == USEC_INFINITY)) {
```

Comment:
i'd drop this `_unlikely_` decorator. on systems lacking a watchdog this is very likely, hence let's avoid the prejudice here, we know too little